### PR TITLE
Adds XYAxis Label Rotation

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith Please Review

Fixes GH-597

Signed-off-by: Mathew Zummo mzummo@gmail.com

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage @dtsnell4 Please review

**Comment by dtsnell4** [comment]
looks good to me

File: demos/test-no-amd.html [hunk 1]
```diff
@@ -28,7 +28,7 @@
     <script src="../bower_components/dagre/index.js"></script>
     <script src="../bower_components/colorbrewer/colorbrewer.js"></script>
     <script src="../bower_components/topojson/topojson.js"></script>
-    <script src="../bower_components/d3-cloud/d3.layout.cloud.js"></script>
+    <script src="../bower_components/d3-cloud/build/d3.layout.cloud.js"></script>
```
Comment (GordonSmith) [inline demos/test-no-amd.html]: Invalid for 1.1.2 version

File: src/chart/XYAxis.js [hunk 2]
```diff
@@ -26,7 +26,8 @@
     XYAxis.prototype.publish("xAxisTypeTimePattern", "%Y-%m-%d", "string", "Time Series Pattern");
     XYAxis.prototype.publish("xAxisDomainLow", "", "string", "X-Axis Low");
     XYAxis.prototype.publish("xAxisDomainHigh", "", "string", "X-Axis High");
-    XYAxis.prototype.publish("xAxisOverlapMode", "stagger", "set", "X-Axis Label Overlap Mode", ["none", "stagger", "hide"]);
+    XYAxis.prototype.publish("xAxisOverlapMode", "rotate", "set", "X-Axis Label Overlap Mode", ["none", "stagger", "hide", "rotate"]);
```
Comment (GordonSmith) [inline src/chart/XYAxis.js]: Should default to stagger

**Comment by GordonSmith** [comment]
Demo URL?

**Comment by mlzummo** [comment]
isnt working with window resize

**Comment by mlzummo** [comment]
https://github.com/masayuki0812/c3/blob/afa5f52ac759e4a1f3d118784bf25a22ba1d2621/src/c3.axis.js#L180

**Pull request closed without merging**